scripts/genappimage.sh

```diff
@@ -0,0 +1,143 @@
+
+if [ -z "$ARCH" ]; then
+  export ARCH="$(arch)"
+fi
+
+APP=NVim
+LOWERAPP="${APP,,}"
+ROOT_DIR="$(git rev-parse --show-toplevel)"
+APP_BUILD_DIR="$ROOT_DIR/build"
+APP_DIR="$APP_BUILD_DIR/$APP.AppDir"
+
+# cd neovim
+GIT_REV="$(git rev-parse --short HEAD)"
+
+# Since neovim increments versions slower than vim, using
+# the commit's date makes more sense.
+VIM_VER="$(date -d "@$(git log -1 --format=%ct)" "+%F")"
+
+
+make install DESTDIR="$APP_DIR"
+
+# Move runtime from /usr/local to /usr
+mv "$APP_DIR"/usr/local/* "$APP_DIR/usr/"
+
+cd "$APP_BUILD_DIR"
+
+wget -q https://github.com/probonopd/AppImages/raw/master/functions.sh -O ./appimage_functions.sh
+. ./appimage_functions.sh
+
+cd "$APP".AppDir
+
+# # Also needs grep for gvim.wrapper
+# cp /bin/grep ./usr/bin
+#
+# # install additional dependencies for python
+# # this makes the image too big, so skip it
+# # and depend on the host where the image is run to fulfill those dependencies
+#
+# #URL=$(apt-get install -qq --yes --no-download --reinstall --print-uris libpython2.7 libpython3.2 libperl5.14 liblua5.1-0 libruby1.9.1| cut -d' ' -f1 | tr -d "'")
+# #wget -c $URL
+# #for package in *.deb; do
+# #    dpkg -x $package .
+# #done
+# #rm -f *.deb
+
+
+########################################################################
+# Copy desktop and icon file to AppDir for AppRun to pick them up
+########################################################################
+
+# Download AppRun and make it executable
+#get_apprun
+
+# get_desktop
+find "${ROOT_DIR}" -name "${LOWERAPP}.desktop" -xdev -exec cp {} "${LOWERAPP}.desktop" \;
+
+find "${ROOT_DIR}" -name "nvim.png" -xdev -exec cp {} "${LOWERAPP}.png" \;
+
+# mkdir -p ./usr/lib/x86_64-linux-gnu
+# copy custom libruby.so 1.9
+# find "$HOME/.rvm/" -name "libruby.so.1.9" -xdev -exec cp {} ./usr/lib/x86_64-linux-gnu/ \; || true
+# add libncurses5
+# find /lib -name "libncurses.so.5" -xdev -exec cp -v -rfL {} ./usr/lib/x86_64-linux-gnu/ \; || true
+
+# copy dependencies
+copy_deps
+
+# Move the libraries to usr/bin
+move_lib
+
+########################################################################
+# Delete stuff that should not go into the AppImage
+########################################################################
+
+# if those libraries are present, there will be a pango problem
+# find . -name "libpango*" -delete
+# find . -name "libfreetype*" -delete
+# find . -name "libX*" -delete
+
+# Delete dangerous libraries; see
+# https://github.com/probonopd/AppImages/blob/master/excludelist
+delete_blacklisted
+
+########################################################################
+# desktopintegration asks the user on first run to install a menu item
+########################################################################
+
+# get_desktopintegration "$LOWERAPP"
+
+########################################################################
+# Determine the version of the app; also include needed glibc version
```

Comment:
Ah, I need to change that comment. It used to add glibc to the version string twice - once there and once in the generate_appimage function in https://github.com/probonopd/AppImages/blob/master/functions.sh